r2dbc-mysql/src/main/java/SimpleStatement.kt

```diff
@@ -0,0 +1,67 @@
+package com.github.jasync.r2dbc.mysql
+
+import io.r2dbc.spi.Result
+import io.r2dbc.spi.Statement
+import io.reactivex.BackpressureStrategy
+import io.reactivex.Flowable
+import org.reactivestreams.Publisher
+import java.util.function.Supplier
+import com.github.jasync.sql.db.Connection as JasyncConnection
+
+class SimpleStatement(private val clientSupplier: Supplier<JasyncConnection>, private val sql: String) : Statement {
+
+    private var isPrepared = false
+    private var params: MutableList<Any?> = mutableListOf()
+
+    override fun add(): Statement {
+        TODO("not implemented")
+    }
+
+    override fun bind(identifier: Any, value: Any): Statement {
+        TODO("not implemented")
+    }
+
+    override fun bind(index: Int, value: Any): Statement {
+        isPrepared = true
+        when {
+            index > params.size -> throw IllegalArgumentException("can't skip params when binding")
```

Comment:
I'd propose to keep bindings in a `Map`-like structure. The SPI does not require an ascending binding order. Callers may bind arguments indexes that do not follow an ordering (`bind(0, …).bind(17, …).bind(1, …).bind(2).bind(16,…)`)

Reply:
and to enforce it on the execution itself?

Reply:
Yeah, that makes sense. Give freedom during the binding time and require proper binding on `execute()` and on `add()` (Note: Add records the current binding and prepares the statement to add another set of bindings so that `execute()` calls the statement for each binding, sort of batching). Implementation-wise you would iterate over bindings and issue a new query via something like `Flowable.fromIterable(bindings).flatMap(it -> executePreparedQuery(it))`.

Reply:
I added it